qcelemental/models/molecule.py

```diff
@@ -438,13 +456,26 @@ def from_data(cls, data, dtype, orient=False, **kwargs):
         Molecule
             A constructed molecule class.
         """
+        if dtype is None:
+            if isinstance(data, str):
+                dtype = "string"
+            elif isinstance(dat, np.array):
+                dtype = "numpy"
+            elif isinstance(data, dict):
+                dtype = "dict"
+            else:
+                raise TypeError("Input type not understood, please supply the 'dtype' kwarg.")
 
-        if dtype == "psi4":
-            input_dict = cls._molecule_from_string_psi4(data)
+        if dtype in ["string", "psi4"]:
```

Comment:
if there's a use case where you only wanted to load pure-Cart psi4 (and exclude zmat input), may want to allow psi4/psi4+/xyz/xyz+ in l446 in docstring above.